File: xuri-rpc-java/src/main/java/com/xuri/rpc/MessageReceiverOptions.java
package com.xuri.rpc;

import java.util.concurrent.ConcurrentHashMap;

/**
 * 每个hostId对应的消息接收配置
 * 对应TS中: export interface MessageReceiverOptions
 */
public class MessageReceiverOptions {
    final ObjectOfProxyManager objectOfProxyManager;
    final RemoteProxyManager runnableProxyManager;
    String hostId;
    final ConcurrentHashMap<String, PendingRequest> requestPendingDict;

    MessageReceiverOptions(String hostId) {
        this.objectOfProxyManager = new ObjectOfProxyManager();
        this.runnableProxyManager = new RemoteProxyManager();
        this.hostId = hostId;
        this.requestPendingDict = new ConcurrentHashMap<>();
    }
}

File: xuri-rpc-java/src/main/kotlin/com/xuri/rpc/MessageReceiverOptions.kt
package com.xuri.rpc

import java.util.concurrent.ConcurrentHashMap

/**
 * 每个hostId对应的消息接收配置
 * 对应TS中: export interface MessageReceiverOptions
 */
class MessageReceiverOptions(
    var hostId: String?
) {
    val objectOfProxyManager = ObjectOfProxyManager()
    val runnableProxyManager = RemoteProxyManager()
    val requestPendingDict = ConcurrentHashMap<String, PendingRequest>()
}
